### ISSUE TITLE
New Rule: no-reference-like-url

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#419]
### Rule details

Warn when URLs are also defined identifiers.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
[Mercury](mercury) is the first planet from the sun.

[mercury]: https://example.com/mercury/
```

### Prior Art

```markdown
[remark-lint-no-reference-like-url](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-reference-like-url)
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

While full URLs for definition identifiers are okay ([https://example.com]: https://example.com), and what looks like an identifier could be an actual URL ([text](alpha)), the more common case is that, assuming a definition [alpha]: https://example.com, then a link [text](alpha) should instead have been [text][alpha].

**Comment by lumirlumir** [comment]
@nzakas Thanks for pointing that out — I’ll keep it in mind.

---

@TKDev7 Thanks for your response.

I’d love to hear more thoughts from the @eslint/eslint-team, as I see both pros and cons to this.

Pros: There’s prior art in `remark-lint-no-reference-like-url`, so adopting this rule could help users transition to ESLint — similar to how we've incorporated rules from Markdownlint and Remark-lint.

Cons: I’m not sure how broadly useful this rule would be. It seems to address only a few edge cases.
